### PR TITLE
Improvement: Post Masonry - Added Pagination option.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
Trello Task - https://trello.com/c/SyE8dfF0/66-pagination

 - Added Lazy Load pagination option with 2 types
    1 - Scroll
    2 - Button

### Types of changes
New feature (non-breaking change which adds functionality) 

### How has this been tested?
 - Checked the Pagination.
 - Checked the Backward Compatibility.

### Checklist:
- [x] My code is tested
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-block-helper.php [hunk 1]
```diff
@@ -2767,6 +2767,55 @@ public static function get_post_masonry_css( $attr, $id ) {
 
 			$t_selectors = self::get_post_tablet_selectors( $attr );
 
+			$selectors[' .uagb-post__load-more-wrap'] = array(
```
Comment (vrundakansara) [inline classes/class-uagb-block-helper.php:2770]: Generate this if pagination is enabled.

File: assets/js/post-carousel.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ( function( $ ) {
```
Comment (vrundakansara) [inline assets/js/post-carousel.js:1]: Let's rename this file to post.js

File: dist/blocks/post/class-uagb-post.php [hunk 3]
```diff
@@ -842,7 +997,68 @@ public function post_pagination() {
...
+		}
 
+		/**
+		 * Render Posts HTML for Masonry Pagination.
+		 *
+		 * @param object $query WP_Query object.
+		 * @param array  $attributes Array of block attributes.
+		 * @since x.x.x
+		 */
+		public function masonry_posts_markup( $query, $attributes ) {
```
Comment (vrundakansara) [inline dist/blocks/post/class-uagb-post.php:1029]: Can we reuse the common markup generation function?

File: classes/class-uagb-block-helper.php (hunk 1 above)
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: assets/js/post-carousel.js (hunk 2 above)
Comment (RajkiranBagal) [reply assets/js/post-carousel.js]: Done

File: dist/blocks/post/class-uagb-post.php (hunk 3 above)
Comment (RajkiranBagal) [reply dist/blocks/post/class-uagb-post.php:1030]: Yes, Done

**Comment by vrundakansara** [comment]
### Issue
- Hide the Border Color when Border is none - [[Screenshot](https://share.getcloudapp.com/mXuApP72)]
- Add Tabs (Normal/Hover) for the Text and Background color.
- Remove the entire typography for Load More Button. Just keep Font Size in px.

**Comment by RajkiranBagal** [comment]
> ### Resolve the below-listed issue:
> * Hide the Border Color when Border is none - [[Screenshot](https://share.getcloudapp.com/mXuApP72)]
> * Add Tabs (Normal/Hover) for the Text and Background color.
> * Remove the entire typography for Load More Button. Just keep Font Size in px.

Done.

**Comment by vrundakansara** [comment]
#### Defaults
- Font Size: 13px
- Text color - none
- Background color - none
- Hover Text color - none
- Hover Background color - none
- Padding - Vertical : 8px | Horizontal: 12px
- Border - Solid | 1px | radius: 2px | Color: none;

@RajkiranBagal - Please add the above defaults for the Load More button

**Pull request merged**